File: Capstone-Project-dev/Capstone-Project-dev/fe/mobile/ios/ReactNative/Services/PermissionManager.swift
import Foundation

@objc class PermissionManager: NSObject {
    static let ROLE_BUSINESS_EMPLOYEE = "BUSINESS_DEPARTMENT_EMPLOYEE"

    @objc static let shared = PermissionManager()

    private let tokenManager = TokenManager.shared

    private override init() {
        super.init()
    }

    func canAccessFullFeatures() -> Bool {
        let currentRole = tokenManager.getUserRole()
        return PermissionManager.ROLE_BUSINESS_EMPLOYEE == currentRole
    }

    func canAccessModule(moduleName: String) -> Bool {
        let publicModules = ["HOME", "AUTH", "PROFILE"]

        if publicModules.contains(moduleName.uppercased()) {
            return true
        }

        return canAccessFullFeatures()
    }
}

File: Capstone-Project-dev/Capstone-Project-dev/fe/mobile/android/app/src/main/java/com/capstone/nawaco/infrastructure/security/PermissionManager.java
package com.capstone.nawaco.infrastructure.security;

import androidx.annotation.NonNull;

import java.util.Arrays;
import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class PermissionManager {
    public static final String ROLE_BUSINESS_EMPLOYEE = "BUSINESS_DEPARTMENT_EMPLOYEE";

    private final TokenManager tokenManager;

    @Inject
    public PermissionManager(TokenManager tokenManager) {
        this.tokenManager = tokenManager;
    }

    public boolean canAccessFullFeatures() {
        var currentRole = tokenManager.getUserRole();
        return ROLE_BUSINESS_EMPLOYEE.equals(currentRole);
    }

    public boolean canAccessModule(@NonNull String moduleName) {
        List<String> publicModules = Arrays.asList("HOME", "AUTH", "PROFILE");

        if (publicModules.contains(moduleName.toUpperCase())) {
            return true;
        }

        return canAccessFullFeatures();
    }
}
